### PR TITLE
[ServiceBus] 2024-01-01

### PR DESCRIPTION
**Pull request by schaudhari6254888**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/AuthorizationRules.json [hunk 1]
```diff
@@ -0,0 +1,1256 @@
...
+            "$ref": "./examples/NameSpaces/SBNameSpaceAuthorizationRuleListAll.json"
+          }
+        },
+        "description": "Gets the authorization rules for a namespace.",
+        "externalDocs": {
+          "url": "https://msdn.microsoft.com/en-us/library/azure/mt639376.aspx"
+        },
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"
```
Comment (ramoka178) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/AuthorizationRules.json:54]: >v1 [](http://example.com/codeflow?start=37&length=2)

upgrade to latest common types across all swaggers and for all common types

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/namespace-preview.json [hunk 2]
```diff
@@ -0,0 +1,1171 @@
+{
```
Comment (ramoka178) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/namespace-preview.json:1]: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/failover is removed. 

Is this going to be added ? 

If users need to use this API, do they have to use old api versions ?

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/namespace-preview.json [hunk 3]
```diff
@@ -0,0 +1,1171 @@
...
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceNamespacePatch"
+        }
+      ],
+      "description": "Description of a namespace resource."
+    },
+    "SBNamespaceProperties": {
```
Comment (ramoka178) [inline specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/namespace-preview.json:692]: >SBNamespaceProperties [](http://example.com/codeflow?start=5&length=21)

geoDataReplication is removed. 

Wont this cause an issue when users upgrade to this API version ?

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/namespace-preview.json (hunk 2 above)
Comment (schaudhari6254888) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/namespace-preview.json:1]: Hey @ramoka178, This api present in preview version. we dont want to put it into stable version of api
So we removed it

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/namespace-preview.json (hunk 3 above)
Comment (schaudhari6254888) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/namespace-preview.json:692]: Same as above this is the part of `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ServiceBus/namespaces/{namespaceName}/failover` api. it is in preview

File: specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/AuthorizationRules.json (hunk 1 above)
Comment (schaudhari6254888) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/AuthorizationRules.json]: Updated

Comment (gary-x-li) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/AuthorizationRules.json]: You are using v1 of common-types. Can you use the latest (v6, or at least v5)? https://github.com/Azure/azure-rest-api-specs/tree/main/specification/common-types/resource-management

Comment (schaudhari6254888) [reply specification/servicebus/resource-manager/Microsoft.ServiceBus/stable/2024-01-01/AuthorizationRules.json]: Updated.

**Review by gary-x-li — APPROVED** [review]

**Review by damodaravadhani — APPROVED** [review]

**Pull request merged**